airflow/dags/libs/news_article/crawler/linkCrawling.py

```
@@ -21,55 +22,73 @@ def get_article_link():
+    weighted_list = [22.5, 19.6, 16.9, 14.4, 12.1, 10.0, 8.1, 6.4, 4.9, 3.6, 2.5, 1.6, 0.9, 0.4, 0.1]
+    crawling_time = datetime.now().strftime('%Y-%m-%d %H:%M:%S')  # 수집 시간은 작업 시작 시 한 번만 기록
 
     all_article_info = [] # 모든 링크들을 저장할 리스트
 
-    for category in categories.keys():
+    for category, category_id in categories.items():
         main_url = f'https://www.itworld.co.kr/{category}/page/1/'
 
         try:
-            r = requests.get(main_url, headers = headers) # requests.get(url)로 요청
+            r = requests.get(main_url) # requests.get(url)로 요청
```

Ruff (0.12.2)

Probable use of `requests` call without timeout

(S113)